webapp/tests/karma/ts/reducers/reports.spec.ts

```diff
@@ -616,89 +700,88 @@ describe('Reports Reducer', () => {
       expect(state).to.deep.equal({
         reports: [],
         reportsById: new Map(),
-        selected: [],
+        selectedReport: undefined,
+        selectedReports: [],
         verifyingReport: true,
         filters: {},
       });
+
       state = reportsReducer(state, Actions.toggleVerifyingReport());
       expect(state).to.deep.equal({
         reports: [],
         reportsById: new Map(),
-        selected: [],
+        selectedReport: undefined,
+        selectedReports: [],
         verifyingReport: false,
         filters: {},
       });
+
       state = reportsReducer(state, Actions.toggleVerifyingReport());
       expect(state).to.deep.equal({
         reports: [],
         reportsById: new Map(),
-        selected: [],
+        selectedReport: undefined,
+        selectedReports: [],
         verifyingReport: true,
         filters: {},
       });
     });
   });
 
-  describe('setFirstSelectedReportDocProperty', () => {
+  describe('setSelectedReportDocProperty', () => {
     it('should work with empty state', () => {
-      state = reportsReducer(state, Actions.setFirstSelectedReportDocProperty({ prop: true }));
+      state = reportsReducer(state, Actions.setSelectedReportDocProperty({ prop: true }));
+
       expect(state).to.deep.equal({
         reports: [],
         reportsById: new Map(),
-        selected: [],
+        selectedReport: undefined,
+        selectedReports: [],
         verifyingReport: false,
         filters: {},
       });
     });
 
-    it('should update the 1st selected doc', () => {
-      state.selected = [
-        { _id: 'doc', doc: { _id: 'doc', field: 1, _rev: 1 }, formatted: { a: 1 } },
-        { _id: 'doc2', doc: { _id: 'doc2', field: 2, _rev: 1 }, formatted: { a: 2 } },
-        { _id: 'doc3', doc: { _id: 'doc3', field: 3, _rev: 1 }, formatted: { a: 3 } },
-      ];
-      const newState = reportsReducer(state, Actions.setFirstSelectedReportDocProperty({ field: 3, other: 1 }));
+    it('should update the selected doc', () => {
```

Comment:
I'm not sure how this is supposed to work right now, but can there be a case when we have `selectedReport` AND `selectedReports` set? 
If yes, I think keeping the `selectedReports` populated in this test would validate that the property is not changed.

Reply:
Oh good catch! 
Yes, `selectReport` and `selectReports` can have values at the same time.
The scenario is when using mobile devices:
- The user can select multiple reports to bulk delete (`selectReports`), at the same time he can open a report just to see its content (`selectReport`) and not to delete it.

Reply:
Oh thanks. Then I think the doc in selected docs should also be updated?

Reply:
To clarify, if a report is in both selctedReport and selectedReports, there's an update action that changes some values in the report doc, that action should also edit the value in `selectedReports` right? My memory says it does not, but I'm unsure now.

Reply:
right, we have 2 update actions one to update the list of `reports` and the other one to update the list of `selectedReports` it updates the expand and loading properties, I will add a code to also update the `selectedReport`

Reply:
I'll make sure to unit test these :)

Reply:
Done